js-codeshift/Dockerfile

```diff
@@ -0,0 +1,10 @@
+# Container image that runs your code
+FROM node:14
+
+RUN npm install -g jscodeshift
+
+# Copies your code file from your action repository to the filesystem path `/` of the container
+COPY warn-to-log.ts /warn-to-log.ts
+
+# Code file to execute when the docker container starts up (`entrypoint.sh`)
+ENTRYPOINT ["jscodeshift", "-t", "./warn-to-log.ts", "./work"]
```

Comment:
I've also created a Dockerfile for jscodeshift in https://github.com/kkom/tools-docker/pull/2

So that you can check out the repo and push a versioned image to a docker hub user/organisation of your choice with a command like `make push username=kkom tool=jscodeshift version=0.7.0`

In an enterprise setup you'd probably do it to some kind of internal container registry, but that's convenient to just play with different tools easily

Reply:
Nice. The docker rebuilding every time you add a transform is quite painful in my demo. I wonder how we can fix this.

Reply:
I'd say maintain an unofficial docker registry of tools, similar to https://www.linuxserver.io/

I'm just going to use https://github.com/kkom/tools-docker/ for my own purposes, but since for now there is no polling to see if a new version is available, I will manually update it as I need it. And of course everyone needs to assess the risk of pulling images from some random `kkom` Docker hub account.

But in all honesty, people can just push the images themselves based on my source code. It won't add any load on Docker Hub, as the layers are deduplicated across accounts based on their hashes.

And anyone can just push a new version whenever they need one. Writing a batch change is an adhoc process anyway.